docs/transpiling.md

```diff
@@ -1,52 +1,53 @@
 # Transpiling
 
-Step definitions and support files can be written in syntax/language that compiles to JavaScript, and just-in-time compiled when you run Cucumber.
+Step definitions and support files can be written in a syntax or language that compiles to JavaScript, and just-in-time compiled when you run Cucumber. The output of the transpiler must match the module format expected by node, and you must use the correct Cucumber directive to import the code, "import" for esm and "require" for CommonJS.
 
 For example, you might want to use [Babel](https://babeljs.io/):
 
 - In a configuration file `{ requireModule: ['@babel/register'] }`
 - On the CLI `$ cucumber-js --require-module @babel/register`
 
-This would mean any support code loaded with the `require` option would be transpiled first.
+This would mean any support code loaded with the `require` option would be transpiled first then loaded into Cucumber.
 
 ## TypeScript
 
-Your `tsconfig.json` should have these `compilerOptions` on:
+Your `tsconfig.json` should have these `compilerOptions`:
 
 ```json
 "allowSyntheticDefaultImports": true,
 "resolveJsonModule": true,
+
 ```
 
-Other than that, a pretty standard TypeScript setup should work as expected.
+Typescript's output must match Cucumber's expected import, and this is controlled by the "module" option in the `tsconfig.json`.  The default is "CommonJS", but if your project is setup to be an ESM project then Typescript will need to output some generation of ES code to Cucumber. If you aren't sure use "ESNext".
 
-You'll also need to specify where your support code is, since `.ts` files won't be picked up by default.
+Cucumber doesn't load `*.ts` files by default. The glob pattern `step-definitions/**/*.ts` will load them from the default step definition location.
```

Comment:
I don't think this line is an improvement. `step-definitions` isn't any of our default paths.